de.metas.swat/de.metas.swat.base/src/main/java/org/adempiere/mm/attributes/api/ILotNumberBL.java

```diff
@@ -44,4 +45,10 @@
 	 */
 	String calculateLotNumber(Date date);
 
+	/**
+	 * @param asi
+	 * @return the value of the LotNo attribute if found, null otherwise
+	 */
+	String getLotNumberAttributeValue(I_M_AttributeSetInstance asi);
```

Comment:
i suggest to add `..OrNull` to the method name

Reply:
done